dual_net.py

```diff
@@ -134,6 +134,40 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, logdir=None, init_from=None, num_steps=1000):
+        if logdir is None:
+            print("Need to run eval with a logdir")
```

Comment:
I think logdir should maybe just be passed into the DualNetTrainer constructor, and then you can have train/eval subdirectories of the logdir.

Reply:
So are you thinking we pass the root logdir to the constructor and let train/eval append the train/eval string?